cdap-docs/developers-manual/source/data-exploration/streams.rst

```diff
@@ -0,0 +1,214 @@
+.. meta::
+    :author: Cask Data, Inc.
+    :copyright: Copyright © 2015 Cask Data, Inc.
+
+.. _stream-exploration:
+
+============================================
+Stream Exploration
+============================================
+
+Streams are the primary method of ingesting real-time data into CDAP.
+It is often useful to be able to examine data in a Stream in an ad-hoc manner through
+SQL-like queries 
+
+Each event in a Stream contains a timestamp, a map of headers, and a body. When a Stream
+is created, a corresponding Hive table is created that allows queries to be run over
+those three columns. Many times, stream event bodies are also structued and have
+a format and schema of their own. For example, event bodies may be comma delimited
+text or Avro encoded binary data. In these cases it is possible to set a format and schema
+on a Stream, enabling more powerful queries.
+
+Let's take a closer look at attaching format and schema on streams.
+
+Stream Format
+-------------
+
+A format defines how the bytes in an event body can be read as some higher level object.
+For example, the CSV (comma separated values) format can read each value in comma delimited text 
+as a separate column of some given type. Each format has 
+a schema that describes the structure of data the format can read. Some formats, like the avro format,
+require a schema to be explicitly given. Other formats, like the CSV format, have a default schema. 
+
+Format is a configuration setting that can be set on a stream. This can be done through the
+RESTful API, or by using the CLI::
+
+  set stream format <stream-id> <format-name> [<schema>] [<settings>]
+  set stream format mystream csv "f1 int, f2 string"
+
+As mentioned above, a format may support different schemas, which will be discussed in more detail
+in the next section.
+
+It is important to note that formats are applied at read time.
+When events are added to a stream, there are no checks performed to enforce that
+all events added to the stream are readable by the format you have set on a stream.
+If any stream event cannot be read by the format you have set, your query will not work.
```

Comment:
Should talk about what sort of error result? If one event is bad, does the whole thing fail? Or does it return partial results? Or does it work until it hits a bad event?